nativelib/src/main/resources/gc/boehm/gc.c

```diff
@@ -13,12 +14,12 @@ void *scalanative_alloc(void *info, size_t size) {
     return (void *)alloc;
 }
 
-void *scalanative_alloc_raw(size_t size) { return GC_malloc(size); }
-
-void *scalanative_alloc_raw_atomic(size_t size) {
-    return GC_malloc_atomic(size);
+void *scalanative_alloc_atomic(void *info, size_t size) {
+    void **alloc = (void **)GC_malloc_atomic(size);
+    memset(alloc, 0, size);
```

Comment:
Previously it was zeroed in https://github.com/scala-native/scala-native/pull/737/files#diff-738f06a13eea511f93ea2f7c590e0320L81. The only difference is that logic has moved to C code now.